### PR TITLE
[WIP] Upgrade to JUnit 5.3-M1

### PR DESCRIPTION
**Pull request by sormuras**
Needs a polishing pass...

### CODE REVIEW

**Comment by sormuras** [comment]
@forax I wanted to pass TesterConf via a constructor parameter but due to the different class loaders the type looks different at run-time. Had to resort to generic map... do you see a more elegant and type-safe way passing arguments to the TestRunner instance?

**Comment by forax** [comment]
Plugins like the packager makes the difference between the parameters of the plugin PackagerConf and the parameters of the underlying command (jar) by using a simple POJO named Jar.

So instead of using a Map, you can declare a non public immutable class with all the fields you need, if it is created in the same package as the runner it should work because this package is visible by both classloaders (i hope).

**Comment by sormuras** [comment]
Compiling it works, running not.

```text
Caused by: java.lang.NoSuchMethodException: com.github.forax.pro.plugin.tester.TesterRunner.<init>(com.github.forax.pro.plugin.tester.TesterFixture)
        at java.base/java.lang.Class.getConstructor0(Class.java:3326)
        at java.base/java.lang.Class.getConstructor(Class.java:2128)
        at com.github.forax.pro.plugin.tester@0.11/com.github.forax.pro.plugin.tester.TesterPlugin.execute(TesterPlugin.java:94)
```

with `TesterFixture.java` as:

```java
class TesterFixture {
  final Path path;
  final boolean parallel;

  TesterFixture(Path path, boolean parallel) {
    this.path = path;
    this.parallel = parallel;
  }
}
```

**Comment by sormuras** [comment]
Trying to load the fixture class with the new class loader now...

**Comment by sormuras** [comment]
...looks good so far. Next commit is almost done.

**Comment by sormuras** [comment]
Remove the `SlowNoopTests.java` show-casing the effect of parallel execution before merge?

File: plugins/tester/src/main/java/com.github.forax.pro.plugin.tester/com/github/forax/pro/plugin/tester/TesterPlugin.java [hunk 1]
```diff
@@ -81,32 +81,39 @@ public int execute(Config config) throws IOException {
...
       return 1;
     }
   }
 
-  private ClassLoader createTestClassLoader(TesterConf tester, Path testPath, String testModuleName) {
+  private Object create(ClassLoader loader, Class<?> type, Object... args) throws IOException {
+    try {
+      var typeClass = loader.loadClass(type.getName());
+      // TODO Assuming first public c'tor with matching parameter types...
+      return typeClass.getConstructors()[0].newInstance(args);
```
Comment (forax) [inline plugins/tester/src/main/java/com.github.forax.pro.plugin.tester/com/github/forax/pro/plugin/tester/TesterPlugin.java]: var constructor = typeClass.getConstructor(Object[].class);
constructor.newInstance(args);
should work !

**Comment by sormuras** [comment]
I'll rebase and merge to `master` this PR as-is. Then you may tune it to your likings locally. This includes leaving the `SlowNoopTests.java` demo in place or pruning it. Same goes for the changes in `test.pro`. Methinks, setting `tester.parallel(true)` is always a good idea.

D'accord?

**Comment by forax** [comment]
yes, i agree.

for parallel, tester is not the only plugin to have that option, i will perhaps promote it to the convention to have a way to disable parallel build for all plugins.

**Pull request merged**